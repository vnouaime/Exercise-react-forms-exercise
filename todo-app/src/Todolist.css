ul {
    text-align: left;
}

li {
    width: 200px;
    margin-bottom: 10px;
}