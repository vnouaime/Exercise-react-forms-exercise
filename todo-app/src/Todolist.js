import React, {useState} from 'react'
import {v4 as uuid} from 'uuid'
import { Todo } from './Todo'
import { NewTodoForm } from './NewTodoForm'
import './Todolist.css'

/**
 * Returns NewTodoForm and any todo components that have been stored in state
 * 
 * Props: none
 * 
 * State: array of todos
 */

const Todolist = () => {
    const INITIAL_STATE = []

    const [todos, setTodos] = useState(INITIAL_STATE)
    const addTodo = (newTodo) => {
        setTodos(todos => [...todos, {...newTodo, id: uuid()}])
    }
    const deleteTodo = (deleteTodoId) => {
        setTodos(todos => todos.filter(todo => todo.id !== deleteTodoId))
    }

    return (
        <>
        <h1>Add a Todo!</h1>
        <NewTodoForm addTodo={addTodo}/>
        <ul><b>Todo List:</b>
           {todos.map(({ id, todo}) => (
            <li key={id}>
                <Todo 
                key={id} 
                id={id} 
                todo={todo}
                deleteTodo={deleteTodo}
                />
            </li>
            ))} 
        </ul>
        </>
    )
}

export { Todolist }