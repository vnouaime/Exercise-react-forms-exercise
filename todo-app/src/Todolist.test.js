import React from 'react'
import { render, fireEvent } from '@testing-library/react'
import { Todolist } from './Todolist'

function addTodo(todoList, todo="Testing Testing") {
    const todoInput = todoList.getByLabelText("Todo")
    const submitBtn = todoList.queryByText("Add Todo")

    fireEvent.change(todoInput, { target: { value: todoInput}})
    fireEvent.click(submitBtn)
}

test('smoke test of Todolist', () => {
    render(<Todolist />)
})

it("matches snapshot", function() {
    const {asFragment} = render(<Todolist />);
    expect(asFragment()).toMatchSnapshot();
});

it("adds a new todo", function() {
    const todoList = render(<Todolist />)
    expect(todoList.queryByText("X")).not.toBeInTheDocument();
    addTodo(todoList)

    const deleteButton = todoList.getByText("X");
    expect(deleteButton).toBeInTheDocument();
    expect(todoList.queryByText("Testing Testing"))

    expect(todoList.getAllByDisplayValue("")).toHaveLength(1);
})

it("removes a todo", function() {
    const todoList = render(<Todolist />);
    addTodo(todoList);

    const deleteButton = todoList.getByText("X");
    fireEvent.click(deleteButton);
    expect(deleteButton).not.toBeInTheDocument();
})
