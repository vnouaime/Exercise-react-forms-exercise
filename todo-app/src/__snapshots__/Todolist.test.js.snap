// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`matches snapshot 1`] = `
<DocumentFragment>
  <h1>
    Add a Todo!
  </h1>
  <form>
    <label
      for="todo"
    >
      Todo
    </label>
    <input
      id="todo"
      name="todo"
      placeholder="Todo"
      type="text"
      value=""
    />
    <button
      class="NewTodoForm-submitButton"
    >
      Add Todo
    </button>
  </form>
  <ul>
    <b>
      Todo List:
    </b>
  </ul>
</DocumentFragment>
`;
